### PR TITLE
Added Detector Library Samples

### PR DESCRIPTION
**Pull request by aastjais**
*Added Detector Library Samples:*

Added Samples tests for Java, Python, CSharp, Ruby, GoLang, JavaScript, TypeScript, Cloudformation and Terraform.

### CODE REVIEW

**Review by yanjunw-amazon — CHANGES_REQUESTED** [review]
Please do not include `csharp/.DS_Store`

File: cloudformation/src/detectors/disabled-glue-cat-encrypt-cloudformation/non-compliant.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  Resource:
+    Type: 'AWS::Glue::DataCatalogEncryptionSettings'
+    Properties:
+      CatalogId: "CatalogId"
+      DataCatalogEncryptionSettings:
+        ConnectionPasswordEncryption:
+          KmsKeyId: "KmsKeyId"
+          ReturnConnectionPasswordEncrypted: True
+        EncryptionAtRest:
+          # Compliant: Glue Data Catalog Encryption is disabled.
```
Comment (grl-wang) [inline cloudformation/src/detectors/disabled-glue-cat-encrypt-cloudformation/non-compliant.yaml:15]: is this a compliant or non compliant example?

File: golang/src/detectors/http-trace-file-server-as-handler/HTTPTraceFileServerAsHandler.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"log"
+	"net/http"
+)
+
+// {fact rule=http-trace-file-server-as-handler@v1.0 defects=1}
+func httpTraceFileServerAsHandlerNoncompliant() {
+	//Noncompliant: `http.FileServer` used
```
Comment (grl-wang) [inline golang/src/detectors/http-trace-file-server-as-handler/HTTPTraceFileServerAsHandler.go:10]: pls correct the format of the noncompliant comment

```
// Noncompliant: `http.FileServer` used
````

File: golang/src/detectors/http-trace-file-server-as-handler/HTTPTraceFileServerAsHandler.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	keyFile := "YOUR_KEY_FILE"
+	fs := http.FileServer(http.Dir("/dir"))
+	log.Fatal(http.ListenAndServeTLS("192.168.1.101:2000", certFile, keyFile, fs))
+}
+
+// {/fact}
+
+// {fact rule=http-trace-file-server-as-handler@v1.0 defects=0}
+func httpTraceFileServerAsHandlerCompliant() {
+	//Compliant: `http.FileServer` not used
```
Comment (grl-wang) [inline golang/src/detectors/http-trace-file-server-as-handler/HTTPTraceFileServerAsHandler.go:21]: fix the comment format

File: golang/src/detectors/weak-random-number-generation/WeakRandomNumberGeneration.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    return string(result)
+}
+// {/fact}
+
+// {fact rule=weak-random-number-generation@v1.0 defects=0}
+func weakRandomNumberGenerationCompliant(length int) string  {
+	result := make([]byte, length)
+	max := big.NewInt(int64(len(charset)))
+	for i := range result {
+		// Noncompliant: `math.rand` is used for generating random number.
```
Comment (grl-wang) [inline golang/src/detectors/weak-random-number-generation/WeakRandomNumberGeneration.go:29]: any compliant example for this rule?

File: cloudformation/src/detectors/disabled-glue-cat-encrypt-cloudformation/non-compliant.yaml (hunk 1 above)
Comment (aastjais) [reply cloudformation/src/detectors/disabled-glue-cat-encrypt-cloudformation/non-compliant.yaml:15]: comment mistake, will correct this

**Review by grl-wang — APPROVED** [review]

**Pull request merged**